### DISCUSSION TITLE
what is fold level? and why does `za` with foldmethod=expr fold everything without foldlevel=99

### DISCUSSION THREAD

**Discussion by nhtyy** [nvim-treesitter/nvim-treesitter#7285] [Q&A]
without setting foldlevel=99, `za` will fold ALL items

my config is simply setting foldmethod , foldexpr as specified in docs,

This is happening on rust files, happy to share a video or configs thank you

**Comment by clason** [comment]
`:h fold-foldlevel`

**Reply by nhtyy** [reply]
Thank you, maybe my post wasnt clear but Im still not understanding how this description is related to the behavior im experiencing. 

```
'foldlevel' is a number option: The higher the more folded regions are open.
When 'foldlevel' is 0, all folds are closed.
When 'foldlevel' is positive, some folds are closed.
When 'foldlevel' is very high, all folds are open.
'foldlevel' is applied when it is changed.  After that manually folds can be
opened and closed.
When increased, folds above the new level are opened.  No manually opened
folds will be closed.
When decreased, folds above the new level are closed.  No manually closed
folds will be opened.
```

I find it confusing that `za` is described as toggling the fold on your cursor yet it be default folds all possible folds without this option seemingly arbitrarily set to 99

**Reply by nhtyy** [reply]
and for further context and an example, my startup vim script just sets `vim.opt.foldlevel = 99`

without this option however: 
```rust
fn main() { // < -- `za` here folds helper
...
}

fn helper() {
...
}
```
